src/nile/core/account.py

```diff
@@ -152,10 +151,15 @@ async def declare(
         overriding_path=None,
         mainnet_token=None,
         watch_mode=None,
+        nile_account=False,
     ):
         """Declare a contract through an Account."""
         max_fee, nonce, _ = await self._process_arguments(max_fee, nonce)
 
+        if nile_account:
+            assert overriding_path is None
```

Comment:
If it's a nile_account, an overriding_path wouldn't make sense because the path to the artifacts lies within the package. The assertion isn't absolutely necessary, but the user might get the wrong idea about where the account is coming from...an error message with the assertion would be helpful

Reply:
but wouldn't this let users deploy custom accounts?